lib/Core/Memory.cpp

```diff
@@ -476,6 +476,8 @@ ref<Expr> ObjectState::read(unsigned offset, Expr::Width width) const {
   // Treat bool specially, it is the only non-byte sized write we allow.
   if (width == Expr::Bool)
     return ExtractExpr::create(read8(offset), 0, Expr::Bool);
+  if (width == Expr::Int8)
```

Comment:
This looks like a silly microoptimization that makes the code unnecessarily complex. Do you have any data that this is a problem?

Reply:
Think about `mem*` operations, everything is single byte access.

Reply:
Yeah sure, but that loop below doesn't look too different